### PR TITLE
[config] Add new command line to support mellanox sniffer

### PR DESCRIPTION
**Pull request by keboliu**
modified:   config/main.py
	new file:   config/mlnx.py

Signed-off-by:  Kebo Liu kebol@mellanox.com

**- What I did**
Add new command line for Mellanox SDK sniffers
**- How I did it**
 Add new cli group "platform" under "config" command.
 "mlnx" subgroup of "platform" group is for Mellanox platform specific CLI
Add command "config platform mlnx sniffer sdk enable" to enable sdk sniffer
Add command "config platform mlnx sniffer sdk disable" to disable sdk sniffer 
**- How to verify it**
On Mellanox platform can issue these command to enable and disable sdk sniffer.
**- Previous command output (if the output of a command-line utility has changed)**
please refer to design doc [https://github.com/Azure/SONiC/wiki/Mellanox-SDK-and-PRM-Sniffer-Utility-CLI-Design-for-SONiC](url)
**- New command output (if the output of a command-line utility has changed)**
please refer to design doc [https://github.com/Azure/SONiC/wiki/Mellanox-SDK-and-PRM-Sniffer-Utility-CLI-Design-for-SONiC](url)

 This PR needs [sonic-buildimage#1551](https://github.com/Azure/sonic-buildimage/pull/1551) merged first.

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -533,5 +534,15 @@ def ecn(profile, rmax, rmin, ymax, ymin, gmax, gmin, verbose):
...
     run_command(command, display_cmd=verbose)
 
+
+#
+# 'platform' group
+#
+@cli.group()
+def platform():
+    """Platform-related configuration tasks"""
+platform.add_command(mlnx.mlnx)
```
Comment (stcheng) [inline config/main.py]: will this indentation work?

Comment (keboliu) [reply config/main.py]: @stcheng It works well, please refer to line 163 in the same file to see how it load 'aaa' command group.

**Comment by nikos-github** [comment]
@lguohan @keboliu This commit is exposing mlnx specific cli in non mlnx platforms. Please fix.
root@sonic:/# show platform summary
Platform: x86_64-dell_z9100_c2538-r0
HwSKU: Force10-Z9100
ASIC: broadcom

root@sonic:/# show platform ?
Usage: show platform [OPTIONS] COMMAND [ARGS]...

  Show platform-specific hardware info

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  mlnx       Mellanox platform specific configuration...
  psustatus  Show PSU status information
  summary    Show hardware platform information
  syseeprom  Show system EEPROM information
root@sonic:/# show platform mlnx ?
Usage: show platform mlnx [OPTIONS] COMMAND [ARGS]...

  Mellanox platform specific configuration tasks

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  sniffer  Sniffer running status
root@sonic:/# show platform mlnx sniffer
sdk sniffer is OFF

**Comment by nikos-github** [comment]
@keboliu @lguohan Same for config:
root@sonic:/# config platform ?
Usage: config platform [OPTIONS] COMMAND [ARGS]...

  Platform-related configuration tasks

Options:
  --help  Show this message and exit.

Commands:
  mlnx  Mellanox platform specific configuration...

**Comment by keboliu** [comment]
@nikos-github thanks for reporting this issue, added it to todo list.

**Pull request merged**